docs/DeveloperGuide.md

```diff
@@ -1,25 +1,283 @@
+  
+#### Implementation details
+
+The following sequence diagram shows how the `Storage` works.
+
+![StorageSequenceDiagram](./images/StorageSequenceDiagram.png)
+  
+1. At the start of `Duke`, a new `Storage` object will be created.
+2. `Duke` calls loading methods (i.e. `loadTask()`, `loadBook()`, `loadCredit()`, `loadLinks()`, `loadModule()`) 
+sequentially. Each loading method calls the corresponding helper method (i.e. `loadTaskFromLine()`, `loadBookFromLine()`, 
+`loadCreditFromLine()`, `loadLinkFromLine()`, `loadModuleFromLine()`) to load `Item`s from each line in the file. 
+3. After each command, `Duke` calls the `save()` method of `Storage` to save all the `Item`s in the list to files.
 
 
 ## Product scope
 ### Target user profile
-
-{Describe the target user profile}
+Undergraduate students of National University of Singapore who:
+- require help to better manage their school work.
+- forgets to return their loan books to the library on time.
+- wants a timetable planner for easy reference.
+- are lazy to create separate module folders every semester.
+- wish to calculate their CAP.
 
 ### Value proposition
-
-{Describe the value proposition: what problem does it solve?}
+termiNus is an application which helps NUS undergraduates to better manage their school life, by providing daily task or
+borrowed books tracking, and module-related functions. This increase users' efficiency and make their life more organized.
 
 ## User Stories
+|Version|Priority| As a ... | I want to ... | So that I can ...|
+|--------|----------|----------|---------------|------------------|
+|v1.0|***|student|add tasks into a list|keep track of the things I need to do|
+|v1.0|***|student|assign priorities to tasks|focus on the more important things first|
+|v1.0|**|student|assign categories to tasks|have a more organised task list|
+|v1.0|***|student|mark tasks as done|keep track of the remaining tasks to do|
+|v1.0|**|student|list all tasks in my list|have a better overview|
+|v1.0|***|student|be able to delete unwanted tasks|focus on the tasks which I need|
+|v1.0|***|student|save all data after using the application|retrieve the data upon running the application
+|v2.0|**|student|automatically create folders for my modules|I do not have to manually create them|
+|v2.0|***|student|add recurring tasks|avoid adding the same tasks every week
+|v2.0|***|student|have a calendar|I can view my current and upcoming tasks
+|v2.0|***|student|be able to set a tracker my borrowed books|avoid overdue fines
+|v2.0|**|student|sort my tasks based on highest priority|focus on those tasks first
+|v2.0|***|student|save zoom links in a centralized place|easily attend my online classes instead of looking through my email for the link 
+|v2.0|***|student|add modules and calculate my CAP|have a better projection of my grades and efforts
+|v2.0|*|student|login with a password|my system is protected 
+
+## Implementation
```

Comment:
I feel it is a bit confusing that there is another 'Implementation' section here when there were many implementations described in the section above under 'Design & Implementation', maybe this should be moved to there as well? I believe this would go under the Command component.